go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -79,6 +84,10 @@ func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query string, callback func(*sq
 	// FIXME(alainjobart): Add some kind of timeout to the context.
 	ctx := context.Background()
 
+	busyLock.Lock()
+	busy[c] = true
```

Comment:
It looks like nothing cleans up the record from the map until shutdown is called --you need to also delete it when the connection goes away for normal reasons or we will blow up on memory.

Reply:
Oops! Forgot about the delete on `ConnectionClosed`.
The alternate approach of just keeping count would also work. The downside is that we have to return a meaningful error on new requests that will force the client to go elsewhere.

But I agree this may be simpler code. We just have to test and make sure that the error returned will cause the client to go elsewhere.